src/appsettings/validators.py

```diff
@@ -67,3 +69,29 @@ def __call__(self, value):
             if not isinstance(element, self.value_type):
                 params = {"key": key, "value": element, "type": self.value_type.__name__}
                 raise ValidationError(self.message, params=params)
+
+
+class FileValidator(object):
+    """Validator which checks file existence and permissions."""
+
+    DEFAULT_ERROR_MESSAGE = "No required permissions for the file %(value)s."
+
+    def __init__(self, mode, error_message=None):
+        """
+        Validator initialization.
+
+        Args:
+            mode (optional int): Required permission for the file. None means no validation (f.e. file does not exist
+                yet), other values might consist of inclusive OR between F_OK, R_OK, W_OK and X_OK constants. When
+                checking read, write or execute permission, file existence (F_OK) is implicitly required. For more
+                information check ``os.access`` documentation (https://docs.python.org/3/library/os.html#os.access).
+            error_message (str): Override default ``ValidationError`` message. The message must contain ``%(value)s``.
```

Comment:
I thought `ValidationError` wouldn't be satisfied, because `params` attribute would contain invalid values. I probably confused it with another error or string formatter.

Fixed.

Reply:
`ValidationError`s don't care :)